client/concordclient/test/cc_basic_update_queue_test.cpp

```diff
@@ -81,6 +83,42 @@ TEST(trc_basic_update_queue_test, test_release_consumers) {
   consumer1.join();
 }
 
+TEST(trc_basic_update_queue_test, test_pop_with_exception) {
+  auto queue = make_unique<BasicUpdateQueue>();
+  auto test_exception = std::make_exception_ptr(OutOfRangeSubscriptionRequest());
+  queue.reset(new BasicUpdateQueue());
+  EXPECT_NO_THROW(queue->setException(test_exception));
+  EXPECT_THROW(queue->pop();, OutOfRangeSubscriptionRequest);
+
+  queue->push(MakeUniqueUpdate(1, vector<pair<string, string>>{{"key1", "value1"}}));
+  queue->push(MakeUniqueUpdate(2, vector<pair<string, string>>{{"key2", "value2"}}));
+  thread consumer0([&]() { EXPECT_NO_THROW(queue->pop()) << "BasicUpdateQueue::pop call failed with an exception."; });
```

Comment:
No need for threading here. You're testing the exception behaviour of the update queue only. Same for `tryPop`.

```
queue->push()
queue->push()
queue->pop()
queue->setException()
EXPECT_THROW(queue->pop())
EXPECT_NO_THROW(queue->pop())
```

Reply:
Updated.